### PR TITLE
MC-957: add isLightJob flag to JobSummary

### PR DESCRIPTION
**Pull request by zhmaeff**
This change allows MC to filter out light jobs and hide them or show them in a different way on dashboards

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Add `Add to Release Notes` label if changes should be mentioned in release notes or `Not Release Notes content` if changes are not relevant for release notes

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/jet/impl/JobSummary.java [hunk 1]
```diff
@@ -102,6 +118,7 @@ public String getNameOrId() {
      * @deprecated use {@link #getNameOrId()}, semantics is the same
      */
     @Nonnull
+    @Deprecated
```
Comment (viliam-durina) [inline hazelcast/src/main/java/com/hazelcast/jet/impl/JobSummary.java:121]: Why is this deprecated?

File: hazelcast/src/main/java/com/hazelcast/jet/impl/JobSummary.java [hunk 2]
```diff
@@ -52,6 +53,21 @@ public JobSummary(
             @Nonnull JobStatus status,
             long submissionTime
     ) {
+        this(false, jobId, executionId, nameOrId, status, submissionTime);
+    }
+
+    /**
+     * Constructor for a light job
+     */
+    public JobSummary(
```
Comment (viliam-durina) [inline hazelcast/src/main/java/com/hazelcast/jet/impl/JobSummary.java:62]: We don't need to create a new overload, just add the parameter and modify every call site (they are 2 of them).

File: hazelcast/src/main/java/com/hazelcast/jet/impl/JobCoordinationService.java [hunk 3]
```diff
@@ -1201,7 +1205,8 @@ private JobExecutionRecord ensureExecutionRecord(long jobId, JobExecutionRecord
         return record != null ? record : new JobExecutionRecord(jobId, getQuorumSize());
     }
 
-    @SuppressWarnings("WeakerAccess") // used by jet-enterprise
+    @SuppressWarnings("WeakerAccess")
+        // used by jet-enterprise
```
Comment (viliam-durina) [inline hazelcast/src/main/java/com/hazelcast/jet/impl/JobCoordinationService.java:1209]: Did you use autoformat? This looks weird.

Comment (zhmaeff) [reply hazelcast/src/main/java/com/hazelcast/jet/impl/JobCoordinationService.java]: Yes. Reverted all formatting changes.

File: hazelcast/src/main/java/com/hazelcast/jet/impl/JobSummary.java (hunk 1 above)
Comment (zhmaeff) [reply hazelcast/src/main/java/com/hazelcast/jet/impl/JobSummary.java:112]: SonarLint reported missing `@Deprecated` annotation as an issue. You marked this method as deprecated in Java Docs :)

**Review by viliam-durina — APPROVED** [review]

**Review by ufukyilmaz — APPROVED** [review]
Looks good to me

**Review by promanenko — APPROVED** [review]

**Pull request merged**